### DISCUSSION TITLE
Uno v1.0.0

### DISCUSSION THREAD

**Discussion by cvanaret** [cvanaret/Uno#10] [Announcements]
Uno v1.0.0 allows combinations of the following strategies:
- constraint relaxation strategies: l1 relaxation and l1 feasibility restoration.
- subproblems: QP, LP and primal-dual barrier.
- globalization strategies: l1 merit function and filter method (with standard filter or nonmonotone filter).
- globalization mechanisms: line search and trust-region method.

Some combinations are at the moment not fully supported, namely barrier + trust-region (disabled) and l1 merit function + feasibility restoration (Uno issues a warning). This will be improved in later versions of Uno.

The only existing interface is to AMPL ([.nl files](https://en.wikipedia.org/wiki/Nl_(format))).

<hr /><em>This discussion was created from the release <a href='https://github.com/cvanaret/Uno/releases/tag/v1.0.0'>Uno v1.0.0</a>.</em>

**Comment by SanPen** [comment]
Is there an API to create UNO problems? or is it still through AMPL?

**Reply by cvanaret** [reply]
At the moment, there's only an interface to AMPL. I'll update the announcement ;)

**Reply by david0oo** [reply]
We implemented an UNO interface to Casadi such that you could model your problems in Python or Matlab, but it is not 100% finished yet.

**Reply by cvanaret** [reply]
@SanPen we now have C, Julia, Python, and Fortran interfaces :) Check out [the latest release](https://github.com/cvanaret/Uno/releases).

**Reply by SanPen** [reply]
How is the casadi + uno working?
Is there any example?

**Reply by cvanaret** [reply]
> How is the casadi + uno working? Is there any example?

It's on their TO-DO list: https://github.com/casadi/casadi/issues/4224
